### ISSUE TITLE
Downgrade minimum required version of System.Text.Json to 4.0

### ISSUE THREAD

**Issue by mikepizzo** [OData/odata.net#1878]
### Overview 
When we added support to EdmLib for JSON Metadata serialization, we referenced the System.Text.Json 4.0.1.1 library. The default .NETCore app application references System.Text.Json 4.0, so when a developer creates a .NET Core application that references ODL they will get an error that there is a version mismatch.

This is fixed by updating the nuget reference to System.Text.Json for the project, but that's a poor developer experience.

By downgrading EdmLib dependency to align with the default 4.0.1.0 version we can avoid that poor experience.

### Steps to repo:
Create a new .NET Core console application.
Add a nuget reference to Microsoft.OData.Edm 7.7.1.
Enter the following:
~~~c#
            IEdmModel model;
            IEnumerable<EdmError> errors;
            CsdlReader.TryParse(XmlReader.Create("myMetadata.xml"), out model, out errors);
~~~
Attempt to comple

### Expected Result:
No Error.

Actual Result:
Error CS1705:	Assembly 'Microsoft.OData.Edm' with identity 'Microsoft.OData.Edm, Version=7.7.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' uses 'System.Text.Json, Version=4.0.1.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' which has a higher version than referenced assembly 'System.Text.Json' with identity 'System.Text.Json, Version=4.0.1.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'

**Comment by xuzhg** [comment]
Fixed and merged in 7.7.2 version.

**Comment by jmajoor** [comment]
Not fixed in 7.7.2:
Error	CS1705	Assembly 'Microsoft.OData.Edm' with identity 'Microsoft.OData.Edm, Version=7.7.2.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' uses 'System.Text.Json, Version=4.0.1.2, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' which has a higher version than referenced assembly 'System.Text.Json' with identity 'System.Text.Json, Version=4.0.1.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'

**Comment by xuzhg** [comment]
> Not fixed in 7.7.2:
> Error CS1705 Assembly 'Microsoft.OData.Edm' with identity 'Microsoft.OData.Edm, Version=7.7.2.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' uses 'System.Text.Json, Version=4.0.1.2, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' which has a higher version than referenced assembly 'System.Text.Json' with identity 'System.Text.Json, Version=4.0.1.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'

😱

**Comment by Bramvanelderen10** [comment]
Running against the same problem in 7.7.2 also.

@mikepizzo You mentioned that referencing the "System.Text.Json" nuget package directly solved the issue as workaround. For me that also didn't seem to work. I added the latest System.Text.Json nuget package as reference to the project (4.7.2).
Did you do anything else to get it to work?
